File: clinica-frontend/src/types/specialty.ts
export interface Specialty {
  id: number;
  name: string;
  description: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface SpecialtyPayload {
  name: string;
  description: string | null;
  is_active: boolean;
}

export interface SpecialtiesResponse {
  count: number;
  next: string | null;
  previous: string | null;
  results: Specialty[];
}

File: clinic/models/specialty.py
from django.db import models


class Specialty(models.Model):

    name = models.CharField(
        max_length=100,
        unique=True
    )

    description = models.TextField(
        null=True,
        blank=True
    )

    is_active = models.BooleanField(
        default=True
    )

    is_deleted = models.BooleanField(
        default=False
    )

    inactive_reason = models.TextField(
        null=True,
        blank=True
    )

    deactivated_at = models.DateTimeField(
        null=True,
        blank=True
    )

    created_at = models.DateTimeField(
        auto_now_add=True
    )

    updated_at = models.DateTimeField(
        auto_now=True
    )

    class Meta:
        ordering = ['name']

    def __str__(self):
        return self.name